common/configurable/src/main/java/io/helidon/common/configurable/ThreadPool.java

```diff
@@ -556,6 +623,48 @@ public float getAverageSize() {
         public int getPeakSize() {
             return peakSize.get();
         }
+
+        @Override
+        public int drainTo(Collection<? super Runnable> c) {
+            int i = 0;
+            for (Runnable r = poll(); r != null; i++, r = poll()) {
+                c.add(r);
+            }
+            return i;
+        }
+
+        @Override
+        public int drainTo(Collection<? super Runnable> c, int m) {
+            if (m <= 0) {
+                return 0;
+            }
+
+            Runnable r = poll();
+            int i = 1;
```

Comment:
I think there is an off-by-one bug here. It should be safer to increment `i` just before adding `r`:

```
public int drainTo(Collectio<? super Runnable> c, int m) {
   if (m <= 0) {
      return 0;
   }
   Runnable r = poll();
   int i = 0;
   m--;
   while (m > 0 && r != null) {
      i++;
      c.add(r);
      m--;
      r = poll();
   }
   if (r != null) {
      i++;
      c.add(r);
   }
   return i;
}
```